google/cloud/spanner/client.cc

```diff
@@ -0,0 +1,126 @@
+                                 ReadOptions const& /*read_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<ResultSet> Client::Read(
+    Transaction::SingleUseOptions const& /*transaction_options*/,
+    std::string const& /*table*/, KeySet const& /*keys*/,
+    std::vector<std::string> const& /*columns*/,
+    ReadOptions const& /*read_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<ResultSet> Client::Read(Transaction const& /*transaction*/,
+                                 std::string const& /*table*/,
+                                 KeySet const& /*keys*/,
+                                 std::vector<std::string> const& /*columns*/,
+                                 ReadOptions const& /*read_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<ResultSet> Client::Read(SqlPartition const& /*partition*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<std::vector<SqlPartition>> Client::PartitionRead(
+    Transaction const& /*transaction*/, std::string const& /*table*/,
+    spanner::KeySet const& /*keys*/,
+    std::vector<std::string> const& /*columns*/,
+    ReadOptions const& /*read_options*/,
+    PartitionOptions const& /*partition_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<ResultSet> Client::ExecuteSql(
+    SqlStatement const& /*statement*/,
+    ExecuteSqlOptions const& /*sql_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<ResultSet> Client::ExecuteSql(
+    Transaction::SingleUseOptions const& /*transaction_options*/,
+    SqlStatement const& /*statement*/,
+    ExecuteSqlOptions const& /*sql_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<ResultSet> Client::ExecuteSql(
+    Transaction const& /*transaction*/, SqlStatement const& /*statement*/,
+    ExecuteSqlOptions const& /*sql_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<ResultSet> Client::ExecuteSql(SqlPartition const& /* partition */) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+StatusOr<std::vector<SqlPartition>> Client::PartitionQuery(
+    Transaction const& /*transaction*/, SqlStatement const& /*statement*/,
+    ExecuteSqlOptions const& /*sql_options*/,
+    PartitionOptions const& /*partition_options*/) {
+  return Status(StatusCode::kUnimplemented, "not implemented");
+}
+
+std::vector<StatusOr<spanner_proto::ResultSetStats>> Client::ExecuteBatchDml(
+    Transaction const& /*transaction*/,
+    std::vector<SqlStatement> const& /*statements*/, int64_t /*seqno*/) {
+  return {Status(StatusCode::kUnimplemented, "not implemented")};
+}
+
+// returns a lower bound on the number of modified rows
+StatusOr<int64_t> Client::ExecutePartitionedDml(
```

Comment:
To keep to "standard c++" as much as possible, we've been trying to consistently use `std::int64_t` et al.